pkg/cmd/container/list.go

```diff
@@ -77,18 +78,36 @@ func filterContainers(ctx context.Context, client *containerd.Client, filters []
 		}
 	}
 
+	var wg sync.WaitGroup
+	cMap := make(map[string]string)
+	var mu sync.Mutex
+	// formatter.ContainerStatus(ctx, c) is time consuming so we do it in goroutines and return the container's id with status as a map.
+	// prepareContainers func will use this map to avoid call formatter.ContainerStatus again.
+	for _, c := range containers {
+		wg.Add(1)
+		go func(ctx context.Context, c containerd.Container) {
+			defer wg.Done()
+			cStatus := formatter.ContainerStatus(ctx, c)
+			mu.Lock()
+			if c.ID() != "" {
```

Comment:
is it possible to have an empty id ?

if yes we my move this check even before creating the goroutine